yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1926,28 +2166,53 @@ def off_axis_projection_SPH(np.float64_t[:] px,
                             np.uint8_t[:, :] mask,
                             normal_vector,
                             north_vector,
-                            weight_field=None):
+                            weight_field=None,
+                            depth_set=False):
     # Do nothing in event of a 0 normal vector
     if np.allclose(normal_vector, 0.):
         return
 
-    px_rotated, py_rotated, \
-    rot_bounds_x0, rot_bounds_x1, \
-    rot_bounds_y0, rot_bounds_y1 = rotate_particle_coord(px, py, pz,
-                                                         center, width, normal_vector, north_vector)
-
-    pixelize_sph_kernel_projection(projection_array,
-                                   mask,
-                                   px_rotated,
-                                   py_rotated,
-                                   smoothing_lengths,
-                                   particle_masses,
-                                   particle_densities,
-                                   quantity_to_smooth,
-                                   [rot_bounds_x0, rot_bounds_x1,
-                                    rot_bounds_y0, rot_bounds_y1],
-                                   weight_field=weight_field,
-                                   check_period=0)
+    if depth_set:
+        px_rotated, py_rotated, pz_rotated, \
```

Comment:
We shouldn't fix this here, but is this a case where something like a named tuple or simple data class could simplify our code?

Reply:
yes, I do think that would help. it might be worth fixing here too :) I think it could cut down on some of the code duplication I'm introducing.